frontend/src/views/Enrollment/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { Fragment, useState } from "react";
+import propTypes from 'prop-types';
+import Helmet from 'react-helmet';
+import Grid from '@material-ui/core/Grid';
+import { useParticipantByEmail } from '../../data/participants';
+import AvailabilitySelector from '../../components/AvailabilitySelector';
+import Loading from '../../components/Loading';
+import EnrollmentForm from './EnrollmentForm';
+import PostedDialog from './PostedDialog';
+import { participantType, usePostParticipant } from '../../data/participants';
+
+export default function Enrollment({ role, update }) {
+  const { user } = MENTORING_SETTINGS;
+
+  // load the participant's enrollment from the API, if present
+  const [existing] = useParticipantByEmail(user.email);
+  const initialParticipant = existing.participant ? existing.participant : {
+    // no `id` property, meaning a new participant
+    full_name: `${user.first_name} ${user.last_name}`.trim(),
+    email: user.email,
+    manager: '',
+    manager_email: '',
+    is_mentor: role === 'mentor',
+    is_learner: role === 'learner' || Boolean(update),
+    time_availability: AvailabilitySelector.default(),
+    org: '',
+    org_level: '',
+    time_at_org_level: '',
+    learner_interests: [],
+    mentor_interests: [],
+    track_change: 'maybe', // default for mentors, since the choice is disabled
+    org_chart_distance: '',
+    comments: '',
+  };
+
+  return (
+    <Loading loads={[existing]}>
+      <WithLoadedParticipant update={Boolean(existing.participant)} initialParticipant={initialParticipant} />
+    </Loading>
+  );
+}
+
+Enrollment.propTypes = {
+  role: propTypes.string,
+  update: propTypes.bool,
+};
+
+// Sub-component that renders with an initial participant, once that
+// participant is loaded.
+function WithLoadedParticipant({ update, initialParticipant }) {
+  const [participant, setParticipant] = useState(initialParticipant);
+  const [{loading: postLoading, error: postError}, postParticipant] = usePostParticipant(participant);
+  const [posted, setPosted] = useState(false);
+
+  const handleSubmit = event => {
+    postParticipant().then(
+      () => setPosted(true),
+      // errors are reported via postError, so ignore them here
+      () => {});
+    event.preventDefault();
+  };
+
+  if (postError) {
+    return (
+      <div>
+        <h2>Error</h2>
+        <pre>{postError.toString()}</pre>
+      </div>
+    );
+  }
+
+  return (
+    <Fragment>
+      <Helmet>
+        <title>Mozilla Mentorship Program - Enrollment</title>
+      </Helmet>
+      {posted ? <PostedDialog /> : (
+        <Grid container justify="center">
+          <Grid item>
+            <EnrollmentForm
+              update={update}
+              participant={participant}
+              onParticipantChange={setParticipant}
```

Comment:
Nit: I generally find it easier to read code if the prop names are the same as the variable or function names passed to them. I'm not that familiar with hooks though, so maybe this is treated differently?

Reply:
I *think* this is common with hooks.  The `useState` generally has the form `const [something, setSomething] = useState(..);` so the `setSomething` name seems to be a standard.  And I've seen such setters passed as callbacks like this.  I think `onSomeChange={setSomething}` is the replacement for `onSomeChange={this.handleSomeChange}` in the class-based syntax, so just a new version of the `on`/`handle` duality.